### PR TITLE
[WIP] New version 3.0

### PR DESCRIPTION
**Pull request by Nepomuceno**
Included in this version

Removal of the resource_name_convention
Correct the diff generation to avoid recreation when not needed
Added import support

### CODE REVIEW

**Review by nyuen — COMMENTED** [review]
What's the strategy for the unit test with the new provider?

File: azurecaf/resource_name_schemas.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+				Optional: true,
+				ForceNew: true,
+				Default:  true,
+			},
+		},
+	}
+}
+
+func resourceNameStateUpgradeV2(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+	rawState["use_slug"] = true
```
[the remark this answers is not in the document]
Comment (Nepomuceno) [reply azurecaf/resource_name_schemas.go:204]: This is in use right now and yeah it works with the default value.

**Comment by Nepomuceno** [comment]
> What's the strategy for the unit test with the new provider?

Currently there are unit tests for the only resource in the provider

**Comment by AlexBevan** [comment]
import support, very much appreciated buddy

**Pull request closed without merging**